### PR TITLE
feat: Add source tables, sync group ID, shard to sync summary

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery-issues/issues/2590

### CODE REVIEW

File: cli/cmd/tables_v3.go [hunk 1]
```diff
@@ -24,28 +41,23 @@ func tablesV3(ctx context.Context, sourceClient *managedplugin.Client, sourceSpe
...
 	req := &pluginPb.GetTables_Request{
 		Tables: []string{"*"},
 	}
+
 	if filter == "spec" {
 		req.Tables = sourceSpec.Tables
 		req.SkipTables = sourceSpec.SkipTables
 		req.SkipDependentTables = *sourceSpec.SkipDependentTables
 	}
-	getTablesResp, err := sourcePbClient.GetTables(ctx, req)
```
Comment (erezrokah) [inline cli/cmd/tables_v3.go:35]: Extracted this code so I can use it in `sync_v3.go`

File: cli/cmd/sync_v3.go [hunk 2]
```diff
@@ -604,6 +622,15 @@ func syncConnectionV3(ctx context.Context, syncOptions syncV3Options) (syncErr e
 			DestinationPath:     destinationSpecs[i].Path,
 		}
 
+		if destinationSpecs[i].SyncGroupId != "" {
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply cli/cmd/sync_v3.go:625]: That's due to how we map struct fields to columns and vice versa, `Id->id->ID`, `ID->id->ID`, see https://github.com/cloudquery/cloudquery/blob/1d47795e6edbd9d8a0ddebd1e7afc159460bbc8d/cli/cmd/summary.go#L144

I can use `SyncGroupId` and add an exception to caser maybe

[the remark this answers is not in the document]
Comment (erezrokah) [reply cli/cmd/sync_v3.go:625]: Done in https://github.com/cloudquery/cloudquery/pull/19497/commits/1255ac4f6f8b14f33d8c05d7ab9cd77c03677427

[the remark this answers is not in the document]
Comment (erezrokah) [reply cli/cmd/sync_v3.go:625]: Actually reverted it since the exceptions don't work for `ToPascal` only `ToTitle`

**Review by marianogappa — APPROVED** [review]

File: cli/cmd/sync_test.go [hunk 3]
```diff
@@ -247,48 +293,50 @@ func TestSyncWithSummaryTable(t *testing.T) {
...
+			}
 			cmd.SetArgs(argList)
 			err := cmd.Execute()
 			if tc.err != "" {
 				assert.Contains(t, err.Error(), tc.err)
 			} else {
 				assert.NoError(t, err)
 			}
-			summaries := []syncSummary{}
-			if len(tc.summaryTable) > 0 {
```
Comment (erezrokah) [inline cli/cmd/sync_test.go:210]: This condition is always true in the test cases

File: cli/cmd/sync_test.go [hunk 4]
```diff
@@ -204,17 +204,40 @@ func TestSync(t *testing.T) {
 	}
 }
 
+type syncSummaryTable struct {
```
Comment (erezrokah) [inline cli/cmd/sync_test.go:207]: Needed a separate struct to deserialize the summary table as the schema is different from the `summary-location` file.

The table schema has internal columns prefixed with `_cq` and the `summary-location` JSON doesn't

**Comment by erezrokah** [comment]
Followed up with another commit https://github.com/cloudquery/cloudquery/pull/19497/commits/c1b7805b79bf8d1fbffc79b4d81489bdb4d217d2 since we don't need `SyncGroupID` in the summary table (only the summary file). This led me to fix some tests as they we not loading the summary table file correctly

**Pull request merged**